tck/src/test/java/org/reactivestreams/tck/IdentityProcessorVerificationTest.java

```diff
@@ -115,6 +127,145 @@ public void required_spec104_mustCallOnErrorOnAllItsSubscribersIfItEncountersANo
+            int subscriberCount = subs.size();
+            switch (subscriberCount) {
+              case 0:
+                s.onSubscribe(createSubscription(awaitLatch, s, demand1));
+                break;
+              case 1:
+                s.onSubscribe(createSubscription(awaitLatch, s, demand2));
+                break;
+              default:
+                throw new RuntimeException(String.format("This for-test-purposes-processor supports only 2 subscribers, yet got %s!", subscriberCount));
+            }
+          }
+
+          @Override
+          public void onSubscribe(Subscription s) {
+            this.upstreamSubscription = s;
+          }
+
+          @Override
+          public void onNext(Integer elem) {
+            for (Subscriber<? super Integer> subscriber : subscribers) {
+              try {
+                subscriber.onNext(elem);
+              } catch (Throwable t) {
+                env.flop(t, String.format("Calling onNext on [%s] should not throw! See https://github.com/reactive-streams/reactive-streams-jvm#2.13", subscriber));
+              }
+            }
+          }
+
+          @Override
+          public void onError(Throwable t) {
+            for (Subscriber<? super Integer> subscriber : subscribers) {
+              try {
+                subscriber.onError(t);
+              } catch (Exception ex) {
+                env.flop(ex, String.format("Calling onError on [%s] should not throw! See https://github.com/reactive-streams/reactive-streams-jvm#2.13", subscriber));
+              }
+            }
+          }
+
+          @Override
+          public void onComplete() {
+            for (Subscriber<? super Integer> subscriber : subscribers) {
+              try {
+                subscriber.onComplete();
+              } catch (Exception ex) {
+                env.flop(ex, String.format("Calling onComplete on [%s] should not throw! See https://github.com/reactive-streams/reactive-streams-jvm#2.13", subscriber));
+              }
+            }
+          }
+
+          private Subscription createSubscription(CountDownLatch awaitLatch, final Subscriber<? super Integer> s, final AtomicLong demand) {
+            final MySubscription sub = new MySubscription(awaitLatch, s, demand);
+            subs.add(sub);
+            subscribers.add(s);
+            return sub;
+          }
+
+          final class MySubscription implements Subscription {
+            private final CountDownLatch awaitLatch;
+            private final Subscriber<? super Integer> s;
+            private final AtomicLong demand;
+
+            public MySubscription(CountDownLatch awaitTwoLatch, Subscriber<? super Integer> s, AtomicLong demand) {
+              this.awaitLatch = awaitTwoLatch;
+              this.s = s;
+              this.demand = demand;
+            }
+
+            @Override
+            public void request(final long n) {
+              ex.execute(new Runnable() {
```

Comment:
Certainly. And they will call `upstreamSubscription.request(d)` concurrently on line 228. This is not a problem.

The problem is that I don't see any request coordination. For example, if `Subscriber` S1 requests 5 and `Subscriber` requests 10, the upstream source will emit 15 elements to both S1 and S2, violating the contract.